spec/controllers/idv/sessions_controller_spec.rb

```diff
@@ -16,84 +73,122 @@
     before do
       allow(idv_session).to receive(:clear)
       allow(subject).to receive(:idv_session).and_return(idv_session)
-      controller.user_session['idv/doc_auth'] = flow_session
-      controller.user_session['idv/in_person'] = flow_session
+      controller.user_session['idv/doc_auth'] = { idv_doc_auth_session: true }
+      controller.user_session['idv/in_person'] = { idv_in_person_session: true }
       controller.user_session[:decrypted_pii] = pii
     end
 
-    it 'deletes idv session' do
-      expect(idv_session).to receive(:clear)
-
-      delete :destroy
-
-      expect(controller.user_session['idv/doc_auth']).to be_blank
-      expect(controller.user_session['idv/in_person']).to be_blank
-      expect(controller.user_session[:decrypted_pii]).to be_blank
-    end
+    context 'when inherited proofing' do
+      before do
+        allow(IdentityConfig.store).to receive(:inherited_proofing_enabled).and_return(true)
+        allow(subject).to receive(:inherited_proofing?).and_return true
+      end
```

Comment:
Right. That's a tough call. The only scenario where `user_session['idv/inherited_proofing']` would be populated is when Inherited Proofing (IP) is in play. I wouldn't want to set up a generic test scenario that doesn't exist in the real world (i.e. IDV populating `user_session['idv/inherited_proofing']`). It may also be the case where analytics will be logged that are _IP-specific_ - not necessarily reason _not_ to flatten all this, but I think the former is enough to justify.

Reply:
I think it could make sense to test how this behaves in different flows for a feature spec, but that it's fine to limit the controller tests to the circumstances it is actually concerned with. We're not expecting the controller to do anything different for an inherited proofing session, and it doesn't implement the `inherited_proofing?` method we're stubbing here. I think we should remove the separate contexts and inline the shared examples.

Reply:
That makes sense but the reason the `InheritedProofingConcern` (which would include #inherited_proofing?) isn't included is because we're not logging for Inherited Proofing yet (#log_analytics) but we're doing that in the upcoming sprints. If the `flow/param/analytics_ids` will need to be passed to the existing `idv_start_over` event for Inherited Proofing (IP) _specifically_ as the comment says, it seems to me there will be a need to include `InheritedProofingConcern`, use `#inherited_proofing?`, and test if analytics are being logged specifically for IP and IDV, wouldn't you agree? If `flow/param/analytics_ids` will need to be passed for _both_ IDV _and_ IP, than there's no need for the IP/IDV-specific tests. I know Zach has been some changes to the analytics; I'm not sure if passing the `analytics_id` to the `analytics.idv_start_over` event will be a requirement for both IDV and IP, do you?

Reply:
Looks like if we want to distinguish between events specifically, it's best to send the analytics_id for both IP and IDV, so I'll flatten all this out and the card that deals with analytics for IP will have to deal with it.

Reply:
Done.

Reply:
Thanks, yes, my main point is that I would expect the specs to test the behaviors we expect of the controller as it exists in its current form, not some potential future state which we may not be able to accurately predict.

Can you also inline the shared examples? If and when we have some shared characteristics of the controller's handling of different contexts, I think we could consider how to refactor them at that point, but I don't think we need single-use, single-assertion shared examples.

Reply:
Sure.

Reply:
Done.